docs/source/developers/guide/resources.rst

```diff
@@ -27,3 +27,51 @@
 ************************************
 Additional information and resources
 ************************************
+
+On this page we have listed resources that may be relevant or useful for
+contributors who want to learn more about different parts of Apache Arrow.
+
+.. Annotation Glossary
+.. -------------------
+.. ARROW-15130
```

Comment:
What's this?

Reply:
Annotation glossary is meant to capture different topics included in the Apache Arrow and present a short explanation and, if possible, a link for more information. What could be included: dependencies, memory management, common terms used in the community, etc.

See this comment: https://github.com/apache/arrow/pull/11942#discussion_r770411667

See similar work:
- https://arrow.apache.org/docs/c_glib/arrow-glib/annotation-glossary.html
- https://docs.python.org/3/glossary.html
- https://link.springer.com/content/pdf/bbm%3A978-1-4419-1318-0%2F1.pdf

Reply:
Ah, I see, great idea!  Are we better leaving this placeholder out of here ahead of the release of the first version (with the release I assume)?

Reply:
Sorry, not sure I understand (this part is only a comment/TODO to add the Glossary once the issue ARROW-15130 is closed)? I would not remove it, but could add a TODO to make it clearer?

Reply:
Ah, don't worry about